src/search.c

```diff
@@ -0,0 +1,108 @@
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include <assert.h>
+#include <stddef.h>
+#include <simclist.h>
+#include "search.h"
+#include "match.h"
+
+int find_match(char* line, char* word, int pos_start, int lineNum, list_t* matches) {
+    if (strlen(word) > strlen(line)) return -1;
+
+    char* dup = strdup(line);
+
+    char* line2 = strdup(line);
+    int pos = pos_start;
+    char* token = strtok(line2, " ,.!?\r\t\n");
+    match* foundMatch = NULL;
+
+    while (token != NULL) {
+
+        if (strncmp(token, word, strlen(word)) == 0) {
+
+            /* Config match */
+            foundMatch = match_new(token, lineNum, pos, dup);
+            match_append(foundMatch, matches);
+
+            /* Clean-up */
+            free(dup);
+
+            /* Return */
+            return pos;
+        }
+
+        pos += strlen(token) + 1;
+        token = strtok(NULL, " ,.!?\r\t\n");
+
+    }
+
+    return -1;
+
+}
+
+list_t* parse_file_buffered(FILE* pf, int start_line, int end_line, char* word, list_t* matches) {
+
+    char* line = NULL;
+    int wordlen = strlen(word);
+    size_t len = 0;
+    ssize_t read;
+
+    int found = -1;
```

Comment:
This should be a `bool`. No magic numbers, please.

Reply:
Actually, on a second look, it seems like I cannot use false/true for found, as 0 is an important return number for pos. The structure of the program requires found to be -1 if nothing is found

Reply:
Actually, disregard what I just said. I was mistaken. found will work with false/true